### DISCUSSION TITLE
Trigger catch boundary from resource route?

### DISCUSSION THREAD

**Discussion by jabrks** [remix-run/remix#5304] [Q&A]
I am trying to create a resource route that can be enabled/disabled. I thought this would be possible by throwing a 404 response if a certain condition is not met, although it appears that this does not trigger the catch boundary and instead just returns a plain 404 response. Is there any way to get this to behave as though the route does not exist instead?

**Comment by sergiodxa** [comment]
The catch boundary is only used on UI routes (routes with a component export), for resource routes you can throw a 404 response but it will be similar to return it, basically you have full control of what you return from it, this also means if you want to render an HTML you will have to return/throw a HTML response yourself
